vm/src/stdlib/collections.rs

```diff
@@ -126,15 +126,31 @@ mod _collections {
             Ok(count)
         }
 
-        #[pymethod]
-        fn extend(&self, iter: PyIterable, vm: &VirtualMachine) -> PyResult<()> {
+        fn _extend(&self, iter: PyIterable, vm: &VirtualMachine) -> PyResult<()> {
             // TODO: use length_hint here and for extendleft
             for elem in iter.iter(vm)? {
                 self.append(elem?);
             }
             Ok(())
         }
```

Comment:
My suggestion about this function is:
```suggestion
        fn _extend(&self, iter: impl std::iter::Iterator<Item=PyResult>) -> PyResult<()> {
            for elem in iter {
                self.append(elem?);
            }
            Ok(())
        }
```
Then it will take general rust iterator rather than a PyIterable - which also be able to accept the first branch of next function